### DISCUSSION TITLE
Dynamic Destination for cluster based on JWT token cliams

### DISCUSSION THREAD

**Discussion by hamidrezashahnazari** [dotnet/yarp#2392] [Q&A] [answered]
hi everyone i need to forward request to my backend services based on  jwt token claims. i  stored my Relative URL's as a permissions and after user authentication based on token claim if the user have that permission request will forward to expected backend servce , so now should i use Middlewares or you got a better Idea.thanks for tour help

**Comment by MihaZupan** [comment, marked as answer]
If you're pulling the destination URL from the auth state, you can use direct forwarding in a middleware/controller, see https://microsoft.github.io/reverse-proxy/articles/direct-forwarding.html

**Comment by hamidrezashahnazari** [comment]
thanks for your answer, but if I want to do Authorization operation and then forward the requests , is it a good choice to use  direct forwarding ? and also i need to forward request for gRPC later

**Reply by MihaZupan** [reply]
Sure, you can do auth before forwarding.
